src/dev/jest/ts_transform.ts

```diff
@@ -22,19 +22,29 @@ import { JestConfig, TransformOptions } from 'ts-jest/dist/jest-types';
 
 import { getTsProjectForAbsolutePath } from '../typescript';
 
+const DEFAULT_TS_CONFIG_PATH = require.resolve('../../../tsconfig.json');
+const DEFAULT_BROWSER_TS_CONFIG_PATH = require.resolve('../../../tsconfig.browser.json');
+
 function extendJestConfigJSON(jestConfigJSON: string, filePath: string) {
   const jestConfig = JSON.parse(jestConfigJSON) as JestConfig;
   return JSON.stringify(extendJestConfig(jestConfig, filePath));
 }
 
 function extendJestConfig(jestConfig: JestConfig, filePath: string) {
+  let tsConfigFile = getTsProjectForAbsolutePath(filePath).tsConfigPath;
+
+  // swap ts config file for jest tests
+  if (tsConfigFile === DEFAULT_BROWSER_TS_CONFIG_PATH) {
+    tsConfigFile = DEFAULT_TS_CONFIG_PATH;
+  }
+
   return {
     ...jestConfig,
     globals: {
       ...(jestConfig.globals || {}),
       'ts-jest': {
         skipBabel: true,
-        tsConfigFile: getTsProjectForAbsolutePath(filePath).tsConfigPath,
+        tsConfigFile,
```

Comment:
for whatever reason I thought we were using the browser env.  any ideas if jest still includes jsdom even in node?

in any case that might explain https://github.com/elastic/kibana/pull/21811/files.  I had to manually include the fetch mock browser client because it kept using node-fetch instead of window.fetch.  should we not be setting anything on window there?

Reply:
Yeah, we have jest configured to include jsdom, but that only mocks out some things, mostly the document/DOM related stuff. Things like `fetch()` are not supported. This change is specifically necessary because our browser based config is assuming that the code will be run through webpack before being executed by a browser, which would transpile ESM into webpack requires, but when running the code in jest it is run in node right out of TypeScript, so we need to use the node config to get commonJS

Reply:
An alternative option would be to use a light webpack-like transform on these files after then are put through typescript with our browser-based config, but I don't see any reason for the additional work.